mailet/api/src/main/java/org/apache/mailet/Mail.java

```diff
@@ -274,19 +259,6 @@ default boolean hasSender() {
      * @since Mailet API v2.1
      */
     boolean hasAttributes();
-    
-    /**
-     * Removes the attribute with the given name from this Mail instance.
-     * 
-     * @param name the name of the attribute to be removed
-     * @return the value of the removed attribute, or null
-     *      if there was no such attribute (or if the attribute existed
-     *      and its value was null)
-     * @since Mailet API v2.1
-     * @deprecated see {@link #removeAttribute(AttributeName)
-     */
-    @Deprecated
-    Serializable removeAttribute(String name);
```

Comment:
Don't it break compatibility?

I do not feel comfortable removing these methods.

Reply:
I thought it have been deprecated with 3.2.0, it appears it does not.

Reply:
I believe we should avoid some compatibility breakdown in our central APIs. It's different from removing components noone uses.

Reply:
Well, in a perfect world, we should, but in this case, I think it's relevant to do it since we improve numerous of properties (expressiveness, safety, and so one)

Reply:
Agree, but let's do this in 2 steps: let's deprecate it in this version and remove it in next (and document all this).